Return HTTP response codes as typed errors

suhlig opened the pull request:
This allows to differentiate a dashboard not found from other errors, e.g.

```go
_, err = client.GetDatasource(ctx, dsID)

if errors.Is(err, sdk.ErrNotFound) {
    fmt.Fprintf(os.Stderr, "Creating new datasource %s (id=%d)\n", ds.Name, ds.ID)
}
```

GiedriusS commented on rest-datasource.go:
```diff
@@ -22,7 +22,14 @@ package sdk
+)
+
+var (
+	ErrNotFound      = errors.New("not found")
```
Maybe these could go into `rest-common.go` or something? :thinking:

GiedriusS reviewed (commented):
Great stuff! Maybe you could do this for other REST calls as well? Plus, could you please update README.md that Go >=1.13 is required?

suhlig commented:
Thanks for the feedback. I'll be back with some updates to this PR.

suhlig commented:
I amended my commit with the changes you proposed. Test are green on my workstation; not sure how I could run the GitHub actions.

GiedriusS commented:
It seems like linters are failing. Could you please take a look?

suhlig commented:
Should be fixed now.

GiedriusS commented:
Still, some errors are happening. AFAICT the issue is here:
https://github.com/grafana-tools/sdk/blob/master/rest-dashboard_integration_test.go#L87-L90

We should ignore 404 errors here (: @suhlig could you please take a look?

suhlig commented:
Sure! Thanks for catching these. I ran `go test ./...`; shouldn't that have shown the error?

GiedriusS commented:
We also have integration tests. For that, you need a locally running Grafana, most likely in a Docker container, plus environment variable `GRAFANA_INTEGRATION` set to `1`. You can see how it works [here](https://github.com/grafana-tools/sdk/blob/master/.github/workflows/go.yml).

suhlig commented:
The integration test failure turned out to be a valuable one. I had overlooked a few error cases for  [`delete-dashboard-by-uid`](https://grafana.com/docs/grafana/latest/http_api/dashboard/#delete-dashboard-by-uid) and [`get-dashboard-by-uid`](https://grafana.com/docs/grafana/latest/http_api/dashboard/#get-dashboard-by-uid).

I changed the integration test to consistently use `DeleteDashboardByUID` because this one has the 404 documented, whereas `delete-dashboard-by-slug` (`DELETE /api/dashboards/db/:slug`) as used by `DeleteDashboard` is [deprecated since Grafana 5.0](https://grafana.com/docs/grafana/v6.7/http_api/dashboard/#delete-dashboard-by-slug).

I also added instructions to the README on how to run local tests.

GiedriusS commented:
Thanks, I will take a look tomorrow

suhlig commented on rest-dashboard.go:
```diff
@@ -161,21 +162,27 @@ func (r *Client) getRawDashboard(ctx context.Context, path string) ([]byte, Boar
+		err = dec.Decode(&result)
+		boardBytes = []byte(result.Board)
+	case http.StatusNotFound:
+		err = fmt.Errorf("dashboard with path %q %w", path, ErrNotFound)
```
I just added a second commit that extracts `httpStatusCodeError` as suggested. Feel free to squash both commits if you are happy with the result.

GiedriusS commented on rest-admin.go:
```diff
@@ -14,13 +15,17 @@ func (r *Client) CreateUser(ctx context.Context, user User) (StatusMessage, erro
 		return StatusMessage{}, err
 	}
+	if code != http.StatusOK {
+		return StatusMessage{}, fmt.Errorf("HTTP error %d: returns %s", code, raw)
```
Why not use `httpStatusCodeError` everywhere in all of these functions (in this and others)?

GiedriusS reviewed (approved):
Sorry for the delay, just one suggestion.

suhlig commented:
Bump ... anything I need to do before the PR can be merged?
